src/main/java/Storage.java

```diff
@@ -0,0 +1,115 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Represents the file handling aspect of reading and writing files.
+ */
+public class Storage {
+    String filePath;
+    ArrayList<Task> loadedTasks;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+        this.loadedTasks = new ArrayList<>();
+    }
+
+    /**
+     * Writes task information to the file for permanent storage.
+     * @param filePath path to file
+     * @param textToAdd text to be added into the file
+     * @param isAppending indicates if text is being appended to the file
+     * @throws IOException for input/output errors
+     */
+    private void writeToFile(String filePath, String textToAdd, boolean isAppending) throws IOException {
+        FileWriter fw = new FileWriter(filePath, isAppending);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    /**
+     * Returns an ArrayList of tasks from the stored tasks information upon reading the file.
+     * @return ArrayList of existing tasks
+     * @throws DukeException if file is not found
+     * @throws IOException for input/output errors
+     */
+    public ArrayList<Task> load() throws DukeException, IOException {
+        try {
+            File f = new File(this.filePath);
+            f.getParentFile().mkdir();
```

Comment:
Perhaps variables could be named more explicitly for readability? For example, `File file` instead of `File f`.